spec/delayed_job_sqs_spec.rb

```diff
@@ -91,16 +89,115 @@ def whatever(n, _); tell*n; end
       $fake_sqs.api_fail('send_message')
-      sqs_message.should_not_receive(:delete)
+      sqs_message.should_not_receive(:destroy)
       expect { sqs_job.save }.to raise_error(AWS::SQS::Errors::InvalidAction)
     end
   end
-  
+
+  describe '.batch_delay_jobs' do
+    it 'adds jobs into the buffer' do
+      Delayed::Job.batch_delay_jobs do
+        described_class.enqueue(payload_object: SimpleJob.new)
+        expect(Delayed::Job.buffer[DEFAULT_QUEUE_NAME].first.size).to eq(1)
+        described_class.enqueue(payload_object: SimpleJob.new)
+        expect(Delayed::Job.buffer[DEFAULT_QUEUE_NAME].first.size).to eq(2)
+      end
+    end
+
+    context 'when batch_delay_jobs blocks are nested' do
+      it 'adds all jobs to the same buffer' do
+        Delayed::Job.batch_delay_jobs do
+          described_class.enqueue(payload_object: SimpleJob.new)
+
+          Delayed::Job.batch_delay_jobs do
+            described_class.enqueue(payload_object: SimpleJob.new)
+            expect(Delayed::Job.buffer[DEFAULT_QUEUE_NAME].first.size).to eq(2)
+          end
+        end
+      end
+
+      it 'adds jobs after nested blocks to a newly emptied queue' do
+        Delayed::Job.batch_delay_jobs do
+          described_class.enqueue(payload_object: SimpleJob.new)
+
+          Delayed::Job.batch_delay_jobs do
+            described_class.enqueue(payload_object: SimpleJob.new)
+          end
+          expect(Delayed::Job.buffer).to eq({})
+
+          described_class.enqueue(payload_object: SimpleJob.new)
+          expect(Delayed::Job.buffer[DEFAULT_QUEUE_NAME].first.size).to eq(1)
+        end
+        expect(Delayed::Job.buffer).to eq({})
+      end
+    end
+
+    it 'does not contain anything in buffer outside the transaction' do
+      expect { Delayed::Job.batch_delay_jobs { described_class.enqueue(payload_object: SimpleJob.new) } }.
+          to_not change { Delayed::Job.buffer }.from({})
+    end
+
+    context 'when a transaction has not completed' do
+      it 'has not enqueued any messages' do
+        Delayed::Job.batch_delay_jobs do
+          described_class.enqueue(payload_object: SimpleJob.new)
+          described_class.enqueue(payload_object: SimpleJob.new)
+          expect(AWS::SQS.new.queues.first.visible_messages).to eq(0)
+        end
+      end
+    end
+
+    context 'when a transaction does not complete' do
+      before do
+        begin
+          Delayed::Job.batch_delay_jobs do
+            described_class.enqueue(payload_object: SimpleJob.new)
+            described_class.enqueue(payload_object: SimpleJob.new)
+            fail
+          end
+        rescue
+          nil
+        end
+      end
+      it 'does not send any messages' do
+        expect(AWS::SQS.new.queues.first.visible_messages).to eq(0)
+      end
+
+      it 'empties the buffer' do
+        expect(Delayed::Job.buffer).to eq({})
+      end
+
+      it 'stops buffering' do
+        expect(Delayed::Job.buffering?).to be_falsey
```

Comment:
Can't `be false` be used? It'd be kind of strange that method ending with a `?` returned something else than a boolean.